### PR TITLE
📚 Docs

### LINKED ISSUE #7: Document and demonstrate a Production Deployment
### Because
After we have a Demo environment up, we need one for Production, that looks pretty similar, but with the necessary differences.

### Done looks like
- [x] We can deploy to a Production environment
- [x] The process is well documented in the ~~README (or equivalent)~~ docs

### LINKED ISSUE #30: Setup mkdocs publishing
### Because

The deployment documentation has matured beyond what can be easily read in a single markdown file.

[mkdocs-material](https://squidfunk.github.io/mkdocs-material) will allow us to auto deploy a static markdown site, including copy-able code block, section links, and other expanded markdown functionality.

### Done when

- [x] mkdocs project is generated
- [x] existing documentation is added to site
- [x] GitHub Actions to auto deploy to GitHub pages

### PR DESCRIPTION
**Pull request by mrharpo**
# Documentation
Creates documentation with [mkdocs-material](https://squidfunk.github.io/mkdocs-material/)
- Stack
  - Description of images and terminology
- Setup
  - Setup steps for dev and production
- Develop
  - Development command
  - `./ov` CLI commands descriptions
  - Examples
- Deploy
  - Automatic deploy with `./deploy` script
- Manual
  - Steps for manual deployment

## Versioning
Automatic versioning with [mike](https://github.com/jimporter/mike/)
- Updated .github/workflows/docs.yml to version with mike
- Deployed early docs version as `0.1`
- Deployed updated docs as `0.2` as `latest`
- Shows outdated warning banner if using `!latest`

Closes #30
Closes #7

### CODE REVIEW

File: docs/dev.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+: See [Deploy](/deploy) for detailed usage.
+
+#### `docs`
+
+: Build and run the documentation server, with live change reloading.
+
+#### `i` | `init`
+
+: Initialize a development environment.
```
Comment (foglabs) [inline docs/dev.md:85]: running `./ov i`, the builds worked fine, but then on the npm install step I got this error:

```
 => [ov-wag production 1/2] RUN pip install "gunicorn>=20.1.0,<20.2.0"                                                                                  2.0s
 => [ov-wag production 2/2] COPY . .                                                                                                                    0.1s

unknown shorthand flag: 'i' in -it
```

File: docs/setup.md [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+./ov init
+```
+
+???+ abstract "`./ov init` script"
+
+    This is the equivalent of running:
+
+    ```bash
+
+    git submodule init # (1)
```
[the remark this answers is not in the document]
Comment (mrharpo) [reply docs/setup.md:82]: The `init` should always be fine to re-run,  but it is not needed if the repo was cloned with `--recurse-submodules`.

I could see an argument that `update` should not be included in `init`, in the case that you want a different build environment than what is committed.

[the remark this answers is not in the document]
Comment (mrharpo) [reply docs/setup.md:82]: Just had this issue in diagnosing the other issue below :crystal_ball: :open_mouth:

Fixing now.

File: docs/dev.md (hunk 1 above)
Comment (mrharpo) [reply docs/dev.md:85]: Hmmmm... I don't get that on my machine. I actually get a different error from a fresh clone:

```
=> CACHED [ov-front dev 4/5] COPY package.json .                                                                                                                            0.0s
 => CACHED [ov-front dev 5/5] RUN npm i                                                                                                                                      0.0s
 => CACHED [ov-front production 1/2] COPY ./ .                                                                                                                               0.0s
 => CACHED [ov-front production 2/2] RUN npm run build                                                                                                                       0.0s
WARN[0000] The "UID" variable is not set. Defaulting to a blank string. 
WARN[0000] The "GID" variable is not set. Defaulting to a blank string. 

> postinstall
> remix setup node

sh: remix: not found
npm ERR! code 127
npm ERR! path /var/app
npm ERR! command failed
npm ERR! command sh -c -- remix setup node

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2022-08-12T18_21_25_834Z-debug-0.log
```

This is resolved with the combination of [dockerfile changes in ov-frontend PR#6](https://github.com/WGBH-MLA/ov-frontend/pull/6) plus changing to `target: dev` in [ov-deploy PR#32](https://github.com/WGBH-MLA/ov-deploy/pull/32)

Comment (mrharpo) [reply docs/dev.md:85]: Regarding your error, does the same thing happen when running the `init` commands directly?

Also, what versions of docker and compose?
``` bash
$ docker compose version
Docker Compose version v2.7.0
$ docker version
Version:          20.10.12
```

Comment (foglabs) [reply docs/dev.md:85]: docker: `Version:          20.10.11`
docker compose: `Docker Compose version v2.2.1`

my versions are a bit behind, that could be the problem

running the commands directly:
`./ov build` works fine
`./ov c run -it front npm install` fails with the error I mentioned

I'll see if I can update my docker versions, maybe its a change in parsing options

Comment (foglabs) [reply docs/dev.md:85]: now I get the same remix error as you after upgrading

**Review by afred — APPROVED** [review]
👍 can't really spot any potential bugs in this PR due to lack of experience in how mkdocs works, but it all looks logical, and cost of missed bugs i think is small in this case. So let's ship it!

**Pull request merged**